java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/Application.java

```diff
@@ -108,7 +135,10 @@ public static void main(String[] args) throws BuildStepException, IOException {
     boolean disableSourceBuild = cmd.hasOption("n");
 
     Injector injector = Guice.createInjector(
-        new RootModule(jdkMappings, serverMappings, compatImage, mavenImage, gradleImage,
+        new RootModule(jdkMappings, DEFAULT_JDK_MAPPINGS, serverMappings, DEFAULT_SERVER_MAPPINGS,
```

Comment:
Why not merge `jdkMappings` with `DEFAULT_JDK_MAPPINGS` here, just like you're doing with `compatImage` and `DEFAULT_COMPAT_RUNTIME_IMAGE`? I think it would be more consistent, and you wouldn't have to make as many changes to the `RootModule`.

Reply:
`jdkMappings` is just an array of strings from the command line and there is some more validation that gets done to it in RootModule before it can be merged with the default settings map. We could do it in Application, but that validation function is already in RootModule, so it seems natural to pass both these in.

Reply:
Would it make sense to just express the defaults also as an array of strings to make them more easily mergable with `jdkMappings`?

Reply:
I made the change to merge them in Application. It also solves an issue i had overlooked: previously if a command line setting was formatted different with extra spaces inside we would not match it with the default setting that was actually equivalent and when they get cleaned up (spaces removed) the default setting might actually be kept instead. Now the cleaning up of extra spaces happens before the merging, so that problem is gone.

Reply:
PTAL with these changes now up